android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java

```diff
@@ -175,6 +184,75 @@ public void onShowActivityResult(int resultCode, Intent data, Promise promise) {
 		}
 	}
 
+	@ReactMethod
+	public void pickDirectory(Promise promise) {
+		Activity currentActivity = getCurrentActivity();
+
+		if (currentActivity == null) {
+			promise.reject(E_ACTIVITY_DOES_NOT_EXIST, "Current activity does not exist");
+			return;
+		}
+		this.promise = promise;
+		try {
+			Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+			currentActivity.startActivityForResult(intent, PICK_DIR_REQUEST_CODE, null);
+		} catch (Exception e) {
+			sendError(E_FAILED_TO_SHOW_PICKER, "Failed to create directory picker", e);
+		}
+	}
+
+	private void onPickDirectoryResult(int resultCode, Intent data, Promise promise) {
+		if (resultCode == Activity.RESULT_CANCELED) {
+			sendError(E_DOCUMENT_PICKER_CANCELED, "User canceled directory picker");
+			return;
+		} else if (resultCode != Activity.RESULT_OK) {
+			sendError(E_UNKNOWN_ACTIVITY_RESULT, "Unknown activity result: " + resultCode);
+			return;
+		}
+
+		if (data == null || data.getData() == null) {
+			sendError(E_INVALID_DATA_RETURNED, "Invalid data returned by intent");
+			return;
+		}
+		Uri uri = data.getData();
+
+		WritableMap map = Arguments.createMap();
+		map.putString(FIELD_URI, uri.toString());
+
+		String path = uriToPath(uri);
+		if (path != null) {
```

Comment:
I'm inclined to _always_ return the `path` in the response, even if it may be null. I like that it's consistent and more explicit. WDYT?

Reply:
Yes, makes sense. I guess on the js side there's not much difference between the field being null vs not set at all.